packages/neptune-extension/src/common/utils/path.ts

```diff
@@ -0,0 +1,25 @@
+
+import { PlatformNotebook } from "types/platform";
+
+export async function findNonExistantPath(path: string, platformNotebook: PlatformNotebook) {
+  const parts = path.split('.');
+
+  let counter = 0;
+  let newPath = path;
+
+  while (true) {
+    try {
+      await platformNotebook.assertNotebook(newPath);
+    }
+    catch (error) {
+      // file does not exist
+      break;
+    }
+
+    // file exists - check another name
+    newPath = `${parts[0]} Copy${++counter}.ipynb`;
```

Comment:
IMHO there are couple of wrong assumptions here.
1. You assume there is a single dot sign in a filename. It isn't always the truth.
2. You assume that file has always `.ipynb` extension. This is also wrong.

You can solve it by using `path-browserify` methods:
`const extension = path.extname(path)` will return you the file extension, no matter how many dots are in file name.

Then you can retrieve the filename before extension:
`const fileNameBeforeExt = path.basename(path, extension)`


3. There is not needed `Copy` string. As per our offline discussion format should be `${beforeExtension}${++counter}${extension}`